facebook_connector.py

```diff
@@ -45,8 +52,23 @@ def resolve_user_id(user_id):
     Returns:
         The name of the user if it is able to find it, otherwise the input
     """
-    # TODO (neitsch): This method is not implemented yet.  Use the user_id
-    # (format: 12345678@facebook.com) to get the actual name of the user
+    fb_id_pattern_match = _FB_ID_PATTERN.match(user_id)
+    if fb_id_pattern_match:
+        fb_numeric_id = fb_id_pattern_match.group(1)
+        try:
+            if fb_numeric_id not in _mapped_fb_ids:
+                fb_user_page = urlopen("https://www.facebook.com/{}".format(fb_numeric_id))
```

Comment:
Yes, it worked for me. Just try https://www.facebook.com/4 or with your own ID from https://findmyfbid.com/
And I've looked into this, but unfortunately you need to register as an application with Facebook. In turn you receive a token. However, if the token is available publically, then the application will be blocked. So there is no easy way to access the Facebook API. The obvious workaround is to let every user add their own token (which might also make message scraping easier). I don't think too many people choose to go through the work of setting up a Facebook application just for this purpose.